projects/iio_demo_freeRTOS/src/platform/maxim/main.c

```diff
@@ -0,0 +1,157 @@
+*	on or directly connected to an Analog Devices Inc. component.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+* MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "platform_includes.h"
+#include "common_data.h"
+
+#ifdef IIO_EXAMPLE
+#include "iio_example.h"
+#endif
+
+/* Enables/disables tick-less mode */
+unsigned int disable_tickless = 1;
+
+/* Stringification macros */
+#define STRING(x) STRING_(x)
+#define STRING_(x) #x
+
+/* Console ISR selection */
+#if (CONSOLE_UART == 0)
+#define UARTx_IRQHandler UART0_IRQHandler
+#define UARTx_IRQn UART0_IRQn
+
+#else
+#error "Please update ISR macro for UART CONSOLE_UART"
+#endif
+
+/**
+ * @brief iiod example task
+*/
+
+void iiodTask(void *pvParameters)
+{
+	iio_example_main();
+	while(1) {
+		no_os_mdelay(10);
+	}
+}
+
+/**
+ * @brief LED blinking task for multithreading example
+*/
+
+void blinkingTask(void *pvParameters)
+{
+	struct no_os_gpio_desc *led_pin;
+	no_os_gpio_get(&led_pin, &gpio_ip);
+	no_os_gpio_direction_output(led_pin, NO_OS_GPIO_LOW);
+
+	bool LED_ON = true;
+	while (1) {
+		if (!LED_ON) {
+			no_os_gpio_set_value(led_pin, NO_OS_GPIO_HIGH);
+			LED_ON=true;
+			no_os_mdelay(100);
+		}
+		if (LED_ON) {
+			no_os_gpio_set_value(led_pin, NO_OS_GPIO_LOW);
+			LED_ON=false;
+			no_os_mdelay(100);
+		}
+	}
+}
+
+/**
+ * @brief Creating iio example with blinking led task
+ *
+*/
+
+int create_tasks(void)
+{
+	//Initial wait time
+	for (int i = 0; i < 0x00FFFF; i++) {}
+
+	/* Configure task */
+	if ((xTaskCreate(iiodTask, (const char *)"iiodTask", configIIO_APP_STACK_SIZE,
+			 NULL,
+			 tskIDLE_PRIORITY + 1, NULL) != pdPASS)) {
+		printf("xTaskCreate() failed to create a task.\n");
+	} else if ((xTaskCreate(blinkingTask, (const char *)"blinkingTask",
+				configMINIMAL_STACK_SIZE,
+				NULL,
+				tskIDLE_PRIORITY + 1, NULL) != pdPASS)) {
+		printf("xTaskCreate() failed to create a task.\n");
+	} else {
+		/* Start scheduler */
+		printf("Starting scheduler.\n");
+		vTaskStartScheduler();
+	}
```

Comment:
I am not sure this is the best form for the API calls...also, do we want to start the scheduler if a task creation failed?
I would propose something like:
```
	ret = xTaskCreate(iiodTask, (const char *)"iiodTask", configIIO_APP_STACK_SIZE,
			  NULL,
			  tskIDLE_PRIORITY + 1, NULL);
	if (ret != pdPASS) {
		printf("xTaskCreate() failed to create iiodTask task.\n");
		return ret
	}
	ret = xTaskCreate(blinkingTask, (const char *)"blinkingTask",
			  configMINIMAL_STACK_SIZE,
			  NULL,
			  tskIDLE_PRIORITY + 1, NULL);
	if (ret != pdPASS) {
		printf("xTaskCreate() failed to create blinkingTask task.\n");
		return ret;
	}
```